aiosmtpd/handlers.py

```diff
@@ -80,7 +79,7 @@ def process_message(self, peer, mailfrom, rcpttos, data, **kws):
             add_separator = True
         if add_separator:
             print(file=self.stream)
-        self._print_message_content(peer, data)
+        self._print_message_content(session.peer, message.received_data)
```

Comment:
No, they have different lifetime. Session is created on connection stage, and changed on STARTTLS process as it is RFC requirement to forget all session related information. Envelope is message related information and recreated on RSET and post-data state, when it is time to recieve new message. Thats why I created two classes for that. Maybe it is good idea to just link them by field in Envelope, and pass only one parameter in process_message handler, but I have no strong arguments for that.